### PR TITLE
Administration: GetServerName

### LINKED ISSUE #789: NWNX Admin plugin : Can set session name, but not get it
Placeholder issue for Admin plugin.
It provides a setter to the Session name - but not a getter.

Eg: What if you want to read the current server name, so you can display it in a Custom token in a dialog etc.

### PR DESCRIPTION
**Pull request by Baaleos**
Added a new accessor method to get the server name.
Seemed to make sense that if you can set the server name, you should be able to get it too.

Resolves #789

### CODE REVIEW

**Comment by Baaleos** [comment]
fixes #789

**Comment by Daztek** [comment]
Please update the changelog too :D

**Comment by Baaleos** [comment]
Will do

**Comment by Baaleos** [comment]
Updated change log

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ https://github.com/nwnxee/unified/compare/build8193.6...HEAD
 - Core: `NWNX_CORE_LOG_DATE` variable to add the date in Y-M-D format to the timestamp. Timestamps must be enabled. Default: Off
 - Tweaks: `NWNX_TWEAKS_ITEM_CHARGES_COST_MODE` to revert the cost of items with charges to 1.69 values
 - Core: `NWNX_CORE_NWNX_RESOURCE_DIRECTORY_PATH` variable to override the default path of the UserDirectory/nwnx folder
-
+- Administration: Added GetServerName method to retrieve the servers name as seen by server list.
```
Comment (mtijanic) [inline CHANGELOG.md]: Functions just go in the section below, no need for duplication here.

File: Plugins/Administration/Administration.cpp [hunk 2]
```diff
@@ -264,6 +265,12 @@ Events::ArgumentStack Administration::SetServerName(Events::ArgumentStack&& args
     return Events::Arguments();
 }
 
+Events::ArgumentStack Administration::GetServerName(Events::ArgumentStack&& args)
+{
+    const CExoString serverName = Globals::AppManager()->m_pServerExoApp->GetNetLayer()->GetSessionName();
+    return Events::Arguments(std::string(serverName.m_sString ? serverName.m_sString : ""));
```
Comment (mtijanic) [inline Plugins/Administration/Administration.cpp]: Just `serverName.CStr()`. Probably don't even need the `std::string` part.

File: CHANGELOG.md (hunk 1 above)
Comment (Baaleos) [reply CHANGELOG.md]: corrected changelog

File: Plugins/Administration/Administration.cpp (hunk 2 above)
Comment (Baaleos) [reply Plugins/Administration/Administration.cpp]: using CStr now

File: CHANGELOG.md [hunk 3]
```diff
@@ -26,6 +26,7 @@ https://github.com/nwnxee/unified/compare/build8193.6...HEAD
 - Util: SetInstructionLimit()
 - Area: CreateGenericTrigger()
 - Object: GetPositionIsInTrigger()
+- Administration: GetServerName()
```
Comment (Daztek) [inline CHANGELOG.md]: It's under the wrong header now. Move it to under [Unreleased] :D

File: CHANGELOG.md [hunk 4]
```diff
@@ -45,6 +46,7 @@ N/A
 - Core: POS is now saved for characters who were online when the server was shut down
 - Core: Fixed issue where POS would occasionally get corrupted and print errors to the log
 - ServerLogRedirector: Fixed dropped prefixes when using PrintString()
+- Events: Adjusted Stealth hook to not result in infinite loops if you skip on enter / exit.
```
Comment (Daztek) [inline CHANGELOG.md]: I think this was added already.

File: CHANGELOG.md (hunk 3 above)
Comment (Baaleos) [reply CHANGELOG.md]: Not sure I understand - its showing under the [Unreleased] section, under the 
##### New NWScript Functions

Do you want it added to the ### Added  section?

Comment (Daztek) [reply CHANGELOG.md]: Oh, you probably have to pull the updated changelog.md from upstream

Comment (Baaleos) [reply CHANGELOG.md]: Just rebased now - will do the same on the Faction PR too
Will wait until this one is merged - so I don't have to re-base twice etc

File: Plugins/Administration/NWScript/nwnx_admin.nss [hunk 5]
```diff
@@ -116,6 +116,9 @@ void NWNX_Administration_SetModuleName(string name);
 /// @param name The name to give the server.
 void NWNX_Administration_SetServerName(string name);
 
+/// @brief Returns the server's name as shown to the serverlist.
+void NWNX_Administration_GetServerName();
```
Comment (mtijanic) [inline Plugins/Administration/NWScript/nwnx_admin.nss]: Should return string, not void.

**Pull request merged**